code401challenges/src/main/java/linkedList/LinkedList.java

```diff
@@ -50,12 +50,82 @@ public String toString() {
         return String.valueOf(listValues);
     }
 
-    //Helper function getHead
-    //getHead
-    //Helper function getHeadValue
-    //getHeadValue
+    // Write the following methods for the Linked List class:
+    // .append(value) which adds a new node with the given value to the end of the list
+    public void append(T value) {
+        Node<T> currentNode = this.head;
+
+        //If there are no nodes in the list
+        if (currentNode == null) {
+            Node newNode = new Node(value);
+            currentNode = newNode;
+        }
+
+        //Traverse to the end of the list and then append
+        while (currentNode.next != null) {

```

Comment:
this while loop is an infinite loop. your condition checks if current.next != null, within the loop if current.next becomes null, you add a node to the end of the list, the while loop reiterates and the same thing happens. you should move the logic of adding a new Node out of the if statement and put it after the while loop

Reply:
Going back to this it looks like I caught it since submitting. The updated code is below:
while (currentNode.next != null) {
                currentNode = currentNode.next;
            }
            currentNode.next = new Node(value);